src/playwright-api/takeArchive.ts

```diff
@@ -36,12 +46,43 @@ async function takeArchive(
   });
 
   // Serialize and capture the DOM
-  const domSnapshot: serializedNode = await page.evaluate(dedent`
+  const domSnapshot: SerializedNode = await page.evaluate(dedent`
     ${rrweb};
     rrwebSnapshot.snapshot(document, { noAbsolute: true });
   `);
 
+  // XXX_jwir3: We go through and filter any of these that would have file names that would be too large.
+  const sourceMap: Map<string, string> = new Map<string, string>();
+  if (domSnapshot.childNodes.length !== 0) {
+    shortenFileNameSrc(domSnapshot.childNodes, sourceMap);
+  }
+
   testInfo.attach(name, { contentType, body: JSON.stringify(domSnapshot) });
+
+  return Promise.resolve(sourceMap);
+}
+
+function shortenFileNameSrc(
+  input: any,
```

Comment:
Nitpick -- would `NodeList` or `Node[]` work for the `input` type?

Reply:
`SerializedNode` works, but if I add this as a type, then I have to add `.toString()` to line 72 to get the TS compiler to behave. Thoughts on how I might get around this in a cleaner way?

Reply:
nvm. I found a solution.